abi.md

```diff
@@ -110,6 +110,127 @@ Passive bundles shall be lowered to Verilog packed structs.
 Reference types in ports shall be logically split out from aggregates and named
 as though "Aggregate Type Lowering" was used.
 
+## On Groups
+
+Optional groups are lowered to separate modules that are instantiated via
+SystemVerilog `bind`{.verilog} statements.  Each module that contains group
+instances will produce an additional module whose name is derived from the group
+and nested group instance.  Nested groups are appended.  E.g.,
+`<module-name>$g<group-name>` is the name of a generated module for its first
```

Comment:
actually it is more clear down below in the example that you really are appending all the names to the file.

I am still curious if we are trying to have some sort of namespace. Should we state that groups names need to be unique within a circuit, regardless of nesting? Because otherwise I'm not sure how there would not be ambiguity if you directly put a `Group2` thing into a module?

Reply:
I'll clean up this section in a follow-on.

Reply:
Mentioned below, but this appending avoids the need to have every group name be unique. I think the issue of name collisions is avoided as a `group` has to follow the existing `declgroup` pattern. You can't do the following because `group B` is not declared:

```
circuit Foo:
  declgroup A:
    declgroup B:
  module Foo:
    group B:
```

However, you could do the following:

```
circuit Foo:
  declgroup A:
    declgroup B:
  declgroup B:
  module Foo:
    group B:
```

Reply:
ok so the proper way to do the first example correctly then is this:

```
circuit Foo:
  declgroup A:
    declgroup B:
  module Foo:
    groupA:
      group B:
```

?

Reply:
Maybe this is easier to just state that all groups must be unique in the circuit namespace. Will try this as it's much simpler.

Reply:
> ok so the proper way to do the first example correctly then is this:

That's correct.